torch_geometric/explain/algorithm/utils.py

```diff
@@ -63,6 +63,35 @@ def set_hetero_masks(
                     apply_sigmoid=apply_sigmoid,
                 )
 
+        elif (isinstance(module, (HANConv, HGTConv))):
+            # Skip if explicitly set for skipping
+            if (module.explain is False):
+                continue
+
+            loop_mask_dict = {
+                '__'.join(k): edge_index[0] != edge_index[1]
```

Comment:
This seems not working for sparse tensor right? We can raise an error here for now.

Reply:
I haven't actually properly tested this for sparse tensors yet, but they would probably break here, yes. Any suggestions on the best way to check if the edge_index_dict contains sparse tensors? I think there's some helper function somewhere for this, but I can't find it from a quick read-through of the docs.

Reply:
I think we can check whether the edge index in the dict `isinstance(Tensor)`. If not, we raise an error.
We can also add a test that pass sparse tensor adjacency matrix and check whether the error is raised properly
```python
from torch_sparse import SparseTensor

adj = SparseTensor(row=edge_index[0], col=edge_index[1], value=...,
                   sparse_sizes=(num_nodes, num_nodes))
# value is optional and can be None
```
Thanks!

Reply:
I went ahead and implemented an explicit check that the tensor isn't sparse (rather than the other way around, checking if it's a tensor), since instances of torch.sparse tensors will in fact return True on `isinstance(Tensor)`. I added a parameter to the tests to make sure that things don't work when trying to mask edges while passing a sparse edge index dictionary.